Added missing setter to plugin name to directly use the SetupTriggerConfig

dbeuchler opened the pull request:
The `SetupTriggerConfig` cannot be used directly because the `PluginName` has no setter. Also inherit the class brings no possibility to override the property with a setter, therefore all SetupTriggers need an explicit config implementation which leads to a lot of boilerplate. Other plugins like the `ProductAssignment` or `RecipeAssignment` are providing this.

Toxantron commented:
I totally agree, but this would be breaking. Every config overriding the plugin name, would have a compiler error for missing setter.

dbeuchler commented:
> I totally agree, but this would be breaking. Every config overriding the plugin name, would have a compiler error for missing setter.

You are right, this was my fault! It doesn't make me happy, but I think the only way to integrate this is moving it to the next major. So I changed the target branch and added a redirected property to my base class

````cs
    public override string PluginName => OverridePluginName;

    //ToDo: Removed when solved in MORYX-Factory
    [DataMember, PluginNameSelector(typeof(ISetupTrigger))]
    public string OverridePluginName { get; set; }
````
